src/Microsoft.AspNet.Mvc.TagHelpers/GlobbingUtility.cs

```diff
@@ -0,0 +1,131 @@
+﻿// Copyright (c) Microsoft Open Technologies, Inc. All rights reserved.
+// Licensed under the Apache License, Version 2.0. See License.txt in the project root for license information.
+
+using System;
+using System.Collections.Generic;
+using System.Linq;
+using Microsoft.AspNet.FileProviders;
+using Microsoft.AspNet.Http;
+using Microsoft.AspNet.Razor.Runtime.TagHelpers;
+using Microsoft.Framework.Cache.Memory;
+using Microsoft.Framework.FileSystemGlobbing;
+using Microsoft.Framework.FileSystemGlobbing.Abstractions;
+
+namespace Microsoft.AspNet.Mvc.TagHelpers
+{
+    /// <summary>
+    /// Utility methods for <see cref="ITagHelper"/>'s that support attributes containing file globbing patterns.
+    /// </summary>
+    public class GlobbingUtility
+    {
+        /// <summary>
+        /// Creates a new <see cref="GlobbingUtility"/>.
+        /// </summary>
+        /// <param name="cache">The <see cref="IMemoryCache"/> to cache globbing results in.</param>
+        /// <param name="baseDirectory">The base directory to perform file globbing matches against.</param>
+        /// <param name="fileProvider">
+        ///     The <see cref="IFileProvider"/> used to watch for changes to file globbing results.
+        /// </param>
+        /// <param name="requestPathBase">
+        ///     The base path of the current request (e.g. <see cref="HttpRequest.PathBase"/>).
+        /// </param>
+        public GlobbingUtility(
+            IMemoryCache cache,
+            DirectoryInfoBase baseDirectory,
```

Comment:
Might be nice to also provide an overload that just takes an `IHostingEnvironment` instead of this param and the next. That way you don't have to add the `IHostingEnvironment` extension method (very high bar for adding extension methods in MVC) and it makes it easier for users to utilize the GlobbingUtilty.